### PR TITLE
ENH: adding ability to package extensions into SlicerSALT

### PR DESCRIPTION
**Pull request by hina-shah**

### CODE REVIEW

File: CMake/Superbuild/External_SlicerPackageExtensions.cmake [hunk 1]
```diff
@@ -103,6 +103,44 @@ if(DEFINED ${proj}_CONFIGURE)
...
+# Hence the following special case for the Apple.
+
+    if(APPLE)
+      set(extension_install_dir "${extension_install_dir}/${APPLICATION_NAME}.app/Contents")
+      file(
+        GLOB files
+        RELATIVE ${extension_install_dir}/
+        ${extension_install_dir}/*
+        )
+      set(extensions_dir "unknown")
```
Comment (bpaniagua) [inline CMake/Superbuild/External_SlicerPackageExtensions.cmake]: why does it say unknown?

File: CMake/Superbuild/External_SlicerPackageExtensions.cmake [hunk 2]
```diff
@@ -103,6 +103,44 @@ if(DEFINED ${proj}_CONFIGURE)
...
+      endforeach()
+      set(extension_install_dir "${extension_install_dir}/${extensions_dir}")
+
+      # get to the extension name directory (this is where lib and share directories live)
+      file(
+        GLOB files
+        RELATIVE ${extension_install_dir}/
+        ${extension_install_dir}/*
+        )
+      set(extension_name_dir "unknown")
```
Comment (bpaniagua) [inline CMake/Superbuild/External_SlicerPackageExtensions.cmake]: Same comment than above.

File: CMake/Superbuild/External_SlicerPackageExtensions.cmake (hunk 1 above)
Comment (hina-shah) [inline CMake/Superbuild/External_SlicerPackageExtensions.cmake]: It is just an initialization to something.. It's getting set inside.

File: CMake/Superbuild/External_SlicerPackageExtensions.cmake (hunk 2 above)
Comment (hina-shah) [inline CMake/Superbuild/External_SlicerPackageExtensions.cmake]: It is just an initialization to something.. It's getting set inside.

**Comment by hina-shah** [comment]
@laurapascal : can you try the build now? The updated slicer tag should have the fix for missing cli's.

**Comment by laurapascal** [comment]
It works perfectly now! :)

**Pull request merged**